### PR TITLE
fix(notification/githubStatus): add retries to update github check

### PR DESCRIPTION
**Pull request by xavileon**
So updating the check is more reliable in the event of failures to update the check.

Signed-off-by: Xavi León <xavi.leon@schibsted.com>

### CODE REVIEW

**Comment by xavileon** [comment]
@spinnaker-reviewers PTAL

File: echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/GithubNotificationAgent.java [hunk 1]
```diff
@@ -101,7 +112,12 @@ public void sendNotifications(
 
     GithubStatus githubStatus = new GithubStatus(state, targetUrl, description, context);
     try {
-      githubService.updateCheck("token " + token, content.getRepo(), branchCommit, githubStatus);
+      final String repo = content.getRepo();
+      retrySupport.retry(
+        () -> githubService.updateCheck("token " + token, repo, branchCommit, githubStatus),
+        MAX_RETRY,
+        RETRY_BACKOFF,
+        true);
```
Comment (emjburns) [inline echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/GithubNotificationAgent.java]: We tend to not do exponential backoff in other areas of the code.

**Comment by emjburns** [comment]
LGTM once comments addressed!

File: echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/GithubNotificationAgent.java (hunk 1 above)
Comment (robzienert) [reply echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/GithubNotificationAgent.java]: Agree, I would recommend against using the exponential API.

**Comment by xavileon** [comment]
@emjburns Thanks! Comments addressed in https://github.com/spinnaker/echo/pull/407/commits/4c77baa940fe0e0e021ab5ca0f5acf9c17fa2919

**Pull request merged**